Dockerfile

```diff
@@ -21,9 +21,36 @@ RUN dnf -y install gcc gcc-c++ make cmake curl-devel libxml2-devel hdf5 hdf5-dev
 
 RUN systemctl enable httpd.service
 
+# Install Python 3.11 from source
+RUN dnf -y install openssl-devel bzip2-devel libffi-devel sqlite-devel \
+               zlib-devel readline-devel tk-devel xz-devel wget && \
+    cd /usr/src && \
+    wget https://www.python.org/ftp/python/3.11.6/Python-3.11.6.tgz && \
+    tar xzf Python-3.11.6.tgz && \
+    cd Python-3.11.6 && \
+    ./configure --enable-optimizations --with-ensurepip=install && \
+    make -j "$(nproc)" && \
+    make altinstall && \
+    # Symlink python3.11 / pip3.11 to python3 / pip3
+    ln -sf /usr/local/bin/python3.11 /usr/bin/python3 && \
+    ln -sf /usr/local/bin/pip3.11 /usr/bin/pip3 && \
+    # Upgrade pip and install wheel
+    python3 -m pip install --upgrade pip wheel && \
+    # Clean up source
+    cd / && rm -rf /usr/src/Python-3.11.6*
```

Comment:
Definitely not for this PR but we could simplify this by using an image with Python 3.11 or, better yet, some conda package [manager like pixi](https://pixi.sh/dev/deployment/container/) and reduce the Python version to just another line in the requirements.